eng/pipelines/templates/stages/archetype-cpp-release.yml

```diff
@@ -12,7 +12,7 @@ stages:
       - stage: Release_${{ replace(artifact.Name, '-', '_') }}
         displayName: 'Release ${{artifact.name}}'
         dependsOn: ${{ parameters.DependsOn }}
-        condition:  and(succeeded(), ne(variables['Build.Repository.Name'], 'Azure/azure-sdk-for-cpp'), ne(variables['Skip.AllRelease'], 'true'))
```

Comment:
I think what you have will probably work but for most other languages we check explicitly for the public repo name which looks like what we were trying to do before but had the wrong condition. How did this ever work?

Reply:
@weshaggard how do you know this was the wrong condition? Is there a typo or something else I am not seeing? Can you please explain/walk through why the logic was previously wrong?

Reply:
@ahsonkhan , the condition for the step includes checking that the repository name is NO `Azure/azure-sdk-for-cpp`  ( which looks wrong and makes us wonder how it was returning `true` in  the past)

According to Daniel, there must be a DevOps bug in the past which was making `Build.Repositoty.Name` to return empty or NULL and making the condition return `true`.  It got fixed and the bug here finally appeared.

Reply:
@weshaggard, @ahsonkhan -- I added y'all to the thread where I did the investigation. 

In short: We've had this incorrect code checked in for ~1 year and have been releasing without problem. Something changed on the Azure DevOps side that is now manifesting the bad behavior from the original bad code.